apps/src/lab2/views/dialogs/ShareDialog.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {useCallback, useState} from 'react';
+import Typography from '@cdo/apps/componentLibrary/typography';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon/FontAwesomeV6Icon';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import FocusLock from 'react-focus-lock';
+import moduleStyles from './ShareDialog.module.scss';
+import QRCode from 'qrcode.react';
+import copyToClipboard from '@cdo/apps/util/copyToClipboard';
+import {hideShareDialog} from '@cdo/apps/code-studio/components/shareDialogRedux';
+
+const CopyToClipboardButton: React.FunctionComponent<{shareUrl: string}> = ({
+  shareUrl,
+}) => {
+  const [copiedToClipboard, setCopiedToClipboard] = useState(false);
+
+  const handleCopyToClipboard = () => {
+    copyToClipboard(shareUrl, () => {
+      setCopiedToClipboard(true);
+    });
+  };
+
+  return (
+    <button
+      type="button"
+      onClick={handleCopyToClipboard}
+      className={moduleStyles.copyToClipboard}
+    >
+      <FontAwesomeV6Icon
+        iconName={copiedToClipboard ? 'clipboard-check' : 'clipboard'}
+        iconStyle="thin"
+        className={moduleStyles.copyToClipboardIcon}
+      />
+      Copy link to project
+    </button>
+  );
+};
+
+const ShareDialog: React.FunctionComponent<{shareUrl: string}> = ({
```

Comment:
Mind adding some docs indicating that this is a new refreshed share dialog that supports a subset of share features currently?

Reply:
Good call.